pkg/ebpfsyncer/ebpfsyncer.go

```diff
@@ -82,6 +87,15 @@ func (e *ebpfSingleton) SyncInterfaceIngressRules(
 		}()
 	}
 
+	signal.Notify(sigc, os.Interrupt, syscall.SIGTERM)
```

Comment:
Can you test if you can still sig TERM the process (if this is needed)?
In a standalone test process, I can't ctrl-c / sig TERM the process any more after doing this, but it might behave differently with the Operator SDK

-----------------------------------

I tested this quickly and kill -9 obviously works still, but the TERM signal doesn't shut down the process any more:

~~~
$ cat main.go
package main

import (
	"fmt"
	"os"
	"os/signal"
	"syscall"
	"time"
)

type eStruct struct {
	c interface{}
}

func (e eStruct) resetAll() {
	fmt.Println("Reset: ", e.c)
}

func foo(e eStruct) {
	fmt.Println("foo", e)

	sigc := make(chan os.Signal, 1)
	signal.Notify(sigc, os.Interrupt, syscall.SIGTERM)
	go func(c chan os.Signal) {
		// Wait for a SIGTERM
		<-c
		if e.c != nil {
			e.resetAll()
		}
	}(sigc)
}

func main() {
	e := eStruct{
		c: "test",
	}
	foo(e)
	time.Sleep(15 * time.Second)
	fmt.Println("15 seconds are over, normal shutdown")
}
~~~

~~~
$ ./m & pid=$! ; sleep 2; kill $pid ; wait $pid
[1] 81616
foo {test}
Reset:  test
15 seconds are over, normal shutdown
[1]+  Done                    ./m
~~~


When I add `os.Exit(0)`, then the program will shut down upon receiving the signal:
~~~
 19 func foo(e eStruct) {
   20     fmt.Println("foo", e)
   21 
   22     sigc := make(chan os.Signal, 1)
   23     signal.Notify(sigc, os.Interrupt, syscall.SIGTERM)
   24     go func(c chan os.Signal) {
   25         // Wait for a SIGTERM
   26         <-c
   27         if e.c != nil {
   28             e.resetAll()
   29         }
   30         os.Exit(0)
   31     }(sigc)
   32 }
   33 
~~~

~~~
[akaris@linux test-signal]$ ./m & pid=$! ; sleep 2; kill $pid ; wait $pid
[1] 82337
foo {test}
Reset:  test
[1]+  Done 
~~~

This is not a suggestion to add os.Exit, it's merely a question from my side

Reply:
when u delete ds or delete the pods with something like 
`oc delete pod -l=app=ingress-node-firewall-daemon -n openshift-ingress-node-firewall` SIGTERM will be sent and the controller runtime will be bring down all processes so I don't need to exit the process when the signal is detected I just need to cleanup and left the normal bring down flow continue did this answer ur question ? or I missunderstood ur question ?

Reply:
Yeah my question is for the opposite case: when you run kill $pid, will the application be torn down and will the pod restart. Or if you test the process standalone, can you still CTRL-C and it will shut down

Reply:
will try it out and see if I can do much in separate PR thanks for the review!!!